### DISCUSSION TITLE
add removeAll or clean to queryCache

### DISCUSSION THREAD

**Discussion by sabasayer** [posva/pinia-colada#140] [Questions] [answered]
I would like to clean all data of cache in queries with optional filtering. 

The example scenario is: in a user driven application clean all user related cache after logout.

**Comment by posva** [comment, marked as answer]
You can do this with:

```ts
const queryCache = useQueryCache()
queryCache.getEntries(filters).map(entry => queryCache.remove(entry))
```

**Reply by sabasayer** [reply]
I used that approach and was thinking a single method would be easier to use.

**Reply by posva** [reply]
I believe it's better to have just one method to filter entries and individual methods to manipulate it. Smaller surface API 🙂
